internal/cmd/buildtool/generic.go

```diff
@@ -55,3 +65,22 @@ func genericBuildPackage(deps buildtoolmodel.Dependencies, product *product) {
 
 	runtimex.Try0(shellx.RunEx(defaultShellxConfig(), argv, &shellx.Envp{}))
 }
+
+// genericBuildLibrary is the generic function for building a library.
+func genericBuildLibrary(deps buildtoolmodel.Dependencies, product *product) {
+	deps.GolangCheck()
+	os := deps.GOOS()
+
+	log.Infof("building %s for %s/%s", product.Pkg, os, runtime.GOARCH)
+
+	lib := path.Base(product.Pkg)
+	library, err := generateLibrary(lib, os)
```

Comment:
I think here you should write:

```suggestion
	library := runtimex.Try1(generateLibrary(lib, os))
```